monitor/test_result.py

```diff
@@ -15,12 +17,16 @@ def __init__(self, id, model_unique_id, test_unique_id, status) -> None:
         self.status = status
 
     @staticmethod
-    def create_test_result_from_dict(test_result_dict: dict) -> 'TestResult':
+    def create_test_result_from_dict(test_result_dict: dict) -> Optional['TestResult']:
         test_type = test_result_dict.get('test_type')
-        if test_type == 'dbt_test':
-            return DbtTestResult(**test_result_dict)
-        else:
-            return ElementaryTestResult(**test_result_dict)
+        try:
+            if test_type == 'dbt_test':
+                return DbtTestResult(**test_result_dict)
+            else:
+                return ElementaryTestResult(**test_result_dict)
+        except Exception:
+            logger.error(f"Failed parsing test result - {json.dumps(test_result_dict)}")
+        return None
```

Comment:
IMO its clearer that the return None is due to the exception if its inside the exception part :)